wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/PluginFQNParser.java

```diff
@@ -37,6 +41,12 @@
 @Beta
 public class PluginFQNParser {
 
+  private static final String INCORRECT_PLUGIN_FORMAT_TEMPLATE =
+      "Plugin '%s' has incorrect format. Should be: 'registryURL/publisher/name/version' or 'registryURL/name/version'";
+  private static final Pattern PLUGIN_PATTERN =
```

Comment:
It might be a bit easier to maintain this code after rewriting in the following way:
```java
  private static final String REGISTRY_PATTERN = "(?<registry>(https?://)[-./\\w]+(:[0-9]+)?)/";
  private static final String PUBLISHER_PATTERN = ...;
  private static final String NAME_PATTERN = ...;
  private static final String VERSION_PATTERN = "[-.a-z0-9]+)";
  private static final String ID_PATTERN = ...;
  private static final Pattern PLUGIN_PATTERN =
      Pattern.compile(
          "(" + REGISTRY_PATTERN + ")?(?.../" + VERSION_PATTERN);
```
WDYT?

Reply:
+1, changed